kensu/pyspark/spark_connector.py

```diff
@@ -800,40 +787,38 @@ def add_prop(name, value):
                     logging.info("unexpected issue when patching DataFrame.toPandas: {}".format(traceback.format_exc()))
                 try:
                     logging.info('adding spark env var spark.executorEnv.KSU_DISABLE_PY_COLLECTOR=true'
-                          ' to disable kensu-py collector on executor nodes')
-                    spark.conf.set("spark.executorEnv.KSU_DISABLE_PY_COLLECTOR", 'true')
-                    spark.sparkContext.environment['KSU_DISABLE_PY_COLLECTOR'] = 'true'
+                                 ' to disable kensu-py collector on executor nodes')
+                    spark_session.conf.set("spark.executorEnv.KSU_DISABLE_PY_COLLECTOR", 'true')
+                    spark_session.sparkContext.environment['KSU_DISABLE_PY_COLLECTOR'] = 'true'
                     logging.info('done adding spark.executorEnv.KSU_DISABLE_PY_COLLECTOR')
                 except:
                     import traceback
                     logging.info("unexpected issue when adding spark.executorEnv.KSU_DISABLE_PY_COLLECTOR: {}".format(traceback.format_exc()))
 
+            if use_api_client:
```

Comment:
hmm, this do not seem to be api_client, but the whole kensu-py thingy (see two lines below) which basically is `init_kensu_python_tracker` or `init_kensu_python_collector`  or something.

Reply:
This was suggested by Sammy as the point here is to initialize the API client to trigger the rules, so call our API. But I don't want to use `rules` word there as over time it may be something more.